### PR TITLE
Use `ruff --preview`, drop `flake8`

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**

### CODE REVIEW

File: test/common/testlib.py [hunk 1]
```diff
@@ -1864,7 +1862,7 @@ def check_journal_messages(self, machine=None):
         """Check for unexpected journal entries."""
         machine = machine or self.machine
         # on main machine, only consider journal entries since test case start
-        cursor = (machine == self.machine) and self.journal_start or None
+        cursor = self.jornal_start if machine == self.machine else None
```
Comment (martinpitt) [inline test/common/testlib.py:1865]: I buy an "u"

**Review by martinpitt — CHANGES_REQUESTED** [review]
I like!

**Comment by martinpitt** [comment]
Looks good in general, but a sea of `TypeError: 'NoneType' object is not callable`

**Comment by allisonkarlitskaya** [comment]
> Looks good in general, but a sea of `TypeError: 'NoneType' object is not callable`

I think I was too tired when I did this :sweat_smile:

**Review by martinpitt — APPROVED** [review]
I'm good with this, I figure my taste of "fix future complains ealier" differs from Jelle's. Definitively don't want to override you there, just retracting my -1.

**Comment by allisonkarlitskaya** [comment]
> Running with `ruff --preview` means we will get unstable rules by default, means we will get more work and become beta testers. Is that really what we want? Also the `preview` ruleset is likely to change every release becoming quite some churn.

My take: probably like 95% of rules that end up in `--preview` get promoted to stable at some point, so that's work we'd end up having to do anyway... and as a project, we'll well positioned.  We gate our task container updates, so we get a chance to fix things cleanly on incoming new rules, and at least a couple of us have even contributed to ruff already... and it gives us a chance to have input upstream on new rules before they go stable.

If we end up seeing something we don't like we always have the option to mask out that new rule and argue against it upstream, *before* it lands in stable...

> Is it known when the rules become "stable"? Is it a month, a week? I couldn't figure that out when I was looking into it.

Unfortunately, ruff seems a bit opaque on this point.  There's https://docs.astral.sh/ruff/preview/ but it doesn't say much about the lifecycle/trajectory of a preview rule...

**Review by jelly — APPROVED** [review]

**Comment by jelly** [comment]
Merged, I'll handle the files flake8 removal.

**Pull request merged**